### DISCUSSION TITLE
Add Eloquent driver for Products

### DISCUSSION THREAD

**Discussion by sdussaut** [duncanmcclean/simple-commerce#987] [Feature Requests]
Hi!

I saw there was some discussions around adding Eloquent drivers for customers, orders and products. I know customers and orders exist already (I use them and they work fine!). What about the products one ? Is it still planned ? Maybe for v6 ?

I also know about this package https://github.com/statamic/eloquent-driver, but it seems to be more of a generic all-or-nothing kind of solution (and would obviously overlap with the drivers provided by Simple Commerce). 

Thanks!

**Comment by duncanmcclean** [comment]
No, it's not something I'm planning to build.

There hasn't been a lot of people request it and it would add a lot of extra complexity. 

If you need it, you can either use the Eloquent Driver (which will move all your collections to the database) or you can implement your own repository using SC's product & order repositories for inspiration:

* https://github.com/duncanmcclean/simple-commerce/blob/5.x/src/Products/EntryProductRepository.php
* https://github.com/duncanmcclean/simple-commerce/blob/5.x/src/Orders/EloquentOrderRepository.php

**Reply by sdussaut** [reply]
Thank you for the answer. I'll give the Eloquent Driver a go then.

**Reply by Pluuk** [reply]
Did you manage to get it to work? If so, would you be able to publish how it would work with the eloquent driver?

**Reply by sdussaut** [reply]
Hi @Pluuk !

Yes I did manage to make it work. Pretty sure it can be better/cleaner but it works for me :-)

- Along to Simple Commerce, I added the Eloquent Driver addon and chose to let it manage all the collections and entries.
- If you do so, products are effectively managed in the database with one problem: so are orders and customers too. It seems ok at first glance, but orders numbers are not incremented (I suspect this is because all order data is not in its own table column but rather in the catch-all one).
- I then enabled the database management for orders and customers of Simple Commerce and it worked:
  - all collections are managed by Eloquent driver inside the `collections` table
  - all entries (except orders and customers) are managed by Eloquent driver inside the `entries` table (products included!)
  - order entries are managed by Simple Commerce inside the `orders` table
  - customer entries are managed by Simple Commerce inside the `customers` table

I *know* this is not the prettiest way of doing it but this works for me (and anyway I want to manage all my Statamic installation in the database for backup and scaling purposes).

Hope this helps!
